feat: add Diversified Late Acceptance approach (#1253)
TimefoldAI/timefold-solver: docs/src/modules/ROOT/pages/optimization-algorithms/local-search.adoc

  </localSearch>
----

[#diversifiedLateAcceptance]
== Diversified Late acceptance

> 

Reviewer: I'd add a note that it's a preview feature, with a link here:
https://docs.timefold.ai/timefold-solver/latest/upgrading-timefold-solver/backwards-compatibility#previewFeatures

Reviewer: That section also needs updating with the addition of DLAS.

[#diversifiedLateAcceptanceAlgorithm]
=== Algorithm description

Diversified Late Acceptance is similar to Late Acceptance,
but it offers different acceptance and replacement strategies.
A move is accepted if its score matches the current solution score
or is better than the late score (which is the winning score of a fixed number of steps ago).

Scientific paper: https://arxiv.org/pdf/1806.09328[Diversified Late Acceptance Search by M. Namazi, C. Sanderson, M. A. H. Newton, M. M. A. Polash, and A. Sattar]

[#diversifiedLateAcceptanceConfiguration]
=== Configuration

Simplest configuration:

[source,xml,options="nowrap"]
----
  <localSearch>
    <localSearchType>DIVERSIFIED_LATE_ACCEPTANCE</localSearchType>
  </localSearch>
----

The late elements list is updated

> only if the current solution score is worse than the late score or better than the late score and different from the previous one.

Reviewer: This sentence is a bit complicated. I'd break it down into bullet points - updated if X, or if Y.

The size of the late elements list is typically smaller.

Advanced configuration:

[source,xml,options="nowrap"]
----

>   <localSearch>

Reviewer: You should also show the preview configuration.

Author: Can you please explain what you mean by preview configuration?

  </localSearch>
----

[#diversifiedLateAcceptance]
== Diversified Late acceptance

[#diversifiedLateAcceptanceAlgorithm]
=== Algorithm description

Diversified Late Acceptance is similar to Late Acceptance,
but it offers different acceptance and replacement strategies.
A move is accepted if its score matches the current solution score
or is better than the late score (which is the winning score of a fixed number of steps ago).

> Scientific paper: https://arxiv.org/pdf/1806.09328[Diversified Late Acceptance Search by M. Namazi, C. Sanderson, M. A. H. Newton, M. M. A. Polash, and A. Sattar]

Reviewer: ```suggestion
Diversified Late Acceptance was first proposed in https://arxiv.org/pdf/1806.09328[Diversified Late Acceptance Search by M. Namazi, C. Sanderson, M. A. H. Newton, M. M. A. Polash, and A. Sattar]
```
